system/src/system_control_internal.cpp

```diff
@@ -389,6 +399,19 @@ void system_ctrl_set_result(ctrl_request* req, int result, ctrl_completion_handl
     particle::system::SystemControl::instance()->setResult(req, result, handler, data);
 }
 
+int system_set_control_request_filter(uint16_t reqType, void* reserved) {
+    if (std::find(particle::system::SystemControl::vecCtrlReq.begin(), particle::system::SystemControl::vecCtrlReq.end(), reqType) == particle::system::SystemControl::vecCtrlReq.end()) {
```

Comment:
You will add a duplicated type if the last element in the vector is equal to the given type